genesyscloud/architect_user_prompt/genesyscloud_architect_user_prompt_proxy.go

```diff
@@ -0,0 +1,199 @@
+func (p *architectUserPromptProxy) getArchitectUserPrompt(ctx context.Context, id string, includeMediaUris bool, includeResources bool, language []string) (*platformclientv2.Prompt, *platformclientv2.APIResponse, error, bool) {
+	return p.getArchitectUserPromptAttr(ctx, p, id, includeMediaUris, includeResources, language)
+}
+
+// getAllArchitectUserPrompts retrieves a list of user prompts
+func (p *architectUserPromptProxy) getAllArchitectUserPrompts(ctx context.Context, includeMediaUris bool, includeResources bool, nameArr []string) (*[]platformclientv2.Prompt, *platformclientv2.APIResponse, error, bool) {
+	return p.getAllArchitectUserPromptsAttr(ctx, p, includeMediaUris, includeResources, nameArr)
+}
+
+// updateArchitectUserPrompt updates a user prompt
+func (p *architectUserPromptProxy) updateArchitectUserPrompt(ctx context.Context, id string, body platformclientv2.Prompt) (*platformclientv2.Prompt, *platformclientv2.APIResponse, error) {
+	return p.updateArchitectUserPromptAttr(ctx, p, id, body)
+}
+
+// deleteArchitectUserPrompt deletes a user prompt
+func (p *architectUserPromptProxy) deleteArchitectUserPrompt(ctx context.Context, id string, allResources bool) (*platformclientv2.APIResponse, error) {
+	return p.deleteArchitectUserPromptAttr(ctx, p, id, allResources)
+}
+
+// createArchitectUserPromptResource creates a new user prompt resource
+func (p *architectUserPromptProxy) createArchitectUserPromptResource(ctx context.Context, id string, body platformclientv2.Promptassetcreate) (*platformclientv2.Promptasset, *platformclientv2.APIResponse, error) {
+	return p.createArchitectUserPromptResourceAttr(ctx, p, id, body)
+}
+
+// updateArchitectUserPromptResource updates a user prompt resource
+func (p *architectUserPromptProxy) updateArchitectUserPromptResource(ctx context.Context, id string, languageCode string, body platformclientv2.Promptasset) (*platformclientv2.Promptasset, *platformclientv2.APIResponse, error) {
+	return p.updateArchitectUserPromptResourceAttr(ctx, p, id, languageCode, body)
+}
+
+// getArchitectUserPromptIdByName retrieves a user prompt by name
+func (p *architectUserPromptProxy) getArchitectUserPromptIdByName(ctx context.Context, nameArr []string) (string, *platformclientv2.APIResponse, error, bool) {
+	return p.getArchitectUserPromptIdByNameAttr(ctx, p, nameArr)
+}
+
+func createArchitectUserPromptFn(ctx context.Context, p *architectUserPromptProxy, body platformclientv2.Prompt) (*platformclientv2.Prompt, *platformclientv2.APIResponse, error) {
+	prompt, response, err := p.architectApi.PostArchitectPrompts(body)
+	if err != nil {
+		return nil, response, fmt.Errorf("failed to create architect user prompt: %s", err)
+	}
+	return prompt, response, nil
+}
+
+func getArchitectUserPromptFn(ctx context.Context, p *architectUserPromptProxy, id string, includeMediaUris bool, includeResources bool, language []string) (*platformclientv2.Prompt, *platformclientv2.APIResponse, error, bool) {
```

Comment:
includeMediaUris, includeResources they differ based on where it is called

Reply:
The only place I can find where includeMediaUris and includeResources aren't being set to true is in the delete. If we hard code includeMediaUris and includeResources to be true when we call the api it won't affect performance in the delete since we are expecting a 404 from that call anyway

Reply:
that sounds reasonable.